k8sutil/ssl.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import urllib3
+import ipaddress
+import sys
+
+from six import PY3
+
+prev_match_hostname = None
+
+
+def _to_unicode(obj):
+    if isinstance(obj, str) and sys.version_info < (3,):
+        obj = unicode(obj, encoding='ascii', errors='strict')
+    return obj
+
+
+def _ip_address(str):
+    return ipaddress.ip_address(_to_unicode(str).rstrip())
+
+
+def _match_hostname(cert, hostname):
+    global prev_match_hostname
+    if not prev_match_hostname:
+        raise TypeError("prev_match_hostname should not be None")
+    try:
+        # Divergence from upstream: ipaddress can't handle byte str
+        host_ip = _ip_address(hostname)
+        san = cert.get('subjectAltName', ())
```

Comment:
Is my understanding correct: this commit is fixing match_hostname function of python version < v3, so that if the hostname is an IP address, it can still match.